File: sdk/java-sdk/src/test/java/com/evernym/verity/sdk/utils/UtilTest.java
package com.evernym.verity.sdk.utils;

import com.evernym.verity.sdk.TestHelpers;
import org.json.JSONObject;
import org.junit.jupiter.api.Test;

import static com.evernym.verity.sdk.TestHelpers.unpackForwardMessage;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.fail;

public class UtilTest {

    @Test
    public void testPackMessageForVerityAndUnpackForward() throws Exception {
        Context context = null;
        try {
            context = TestHelpers.getContext();

            JSONObject testMessage = new JSONObject().put("hello", "world");
            byte[] packedMessage = Util.packMessageForVerity(context, testMessage);

            JSONObject unpackedMessage = unpackForwardMessage(context, packedMessage);
            assertEquals(testMessage.toString(), unpackedMessage.toString());

            context.closeWallet();
        } catch(Exception e) {
            e.printStackTrace();
            fail();
        } finally {
            TestHelpers.cleanup(context);
        }
    }

    @Test
    public void testGetMessageTypeComplete() {
        String msgType = Util.EVERNYM_MSG_QUALIFIER + "/credential/0.1/status";
        assertEquals(msgType, Util.getMessageType(Util.EVERNYM_MSG_QUALIFIER, "credential", "0.1", "status"));
    }

    @Test
    public void noop() {
        new JSONObject("{}");
    }
}

File: sdk/dotnet-sdk/src/VeritySDK.Test/Utils/UtilTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Json;
using VeritySDK.Transports;
using VeritySDK.Utils;

namespace VeritySDK.Test
{
    [TestClass]
    public class UtilTest
    {

        [TestMethod]
        public void testPackMessageForVerityAndUnpackForward()
        {
            Context context = null;
            try
            {
                context = TestHelpers.getContext();

                JsonObject testMessage = new JsonObject();
                testMessage.Add("hello", "world");
                byte[] packedMessage = Util.packMessageForVerity(context, testMessage);

                JsonObject unpackedMessage = TestHelpers.unpackForwardMessage(context, packedMessage);
                Assert.AreEqual(testMessage.ToString(), unpackedMessage.ToString());

                context.CloseWallet();
            }
            catch (Exception e)
            {
                Assert.Fail(e.Message);
            }
            finally
            {
                TestHelpers.cleanup(context);
            }
        }

        [TestMethod]
        public void testGetMessageTypeComplete()
        {
            String msgType = Util.EVERNYM_MSG_QUALIFIER + "/credential/0.1/status";
            Assert.AreEqual(msgType, Util.getMessageType(Util.EVERNYM_MSG_QUALIFIER, "credential", "0.1", "status"));
        }

        [TestMethod]
        public void noop()
        {
            JsonObject.Parse("{}");
        }
    }
}
